Add security threat submission beta api docs (#16242)
microsoftgraph/microsoft-graph-docs: api-reference/beta/api/security-emailthreatsubmission-post-emailthreats.md

---
title: "Create emailThreatSubmission"
description: "Create a new emailThreatSubmission object."
author: "caigen"
ms.localizationpriority: medium
ms.prod: "security"
doc_type: apiPageType
---

# Create emailThreatSubmission
Namespace: microsoft.graph.security

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

Create a new [emailThreatSubmission](../resources/security-emailthreatsubmission.md) object.

## Permissions
One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

|Permission type|Permissions (from least to most privileged)|
|:---|:---|
|Delegated (work or school account)|ThreatSubmission.ReadWrite, ThreatSubmission.ReadWrite.All|
|Delegated (personal Microsoft account)|N/A|
|Application|ThreatSubmission.ReadWrite.All|

## HTTP request

<!-- {
  "blockType": "ignored"
}
-->
``` http
POST /security/threatSubmission/emailThreats
```

## Request headers
|Name|Description|
|:---|:---|
|Authorization|Bearer {token}. Required.|
|Content-Type|application/json. Required.|

## Request body

> In the request body, supply a JSON representation of the [emailContentThreatSubmission](../resources/security-emailcontentthreatsubmission.md) object or the [emailUrlThreatSubmission](../resources/security-emailurlthreatsubmission.md)

Reviewer: This is a CRUD operation to create emailThreatSubmission resources, please clarify why are also using it to create another resource (emailContentThreatSubmission). Different entities should have their own CRUD operations AFAIK.

## Response

If successful, this method returns a `201 Created` response code and an [emailThreatSubmission](../resources/security-emailthreatsubmission.md) object in the response body.

## Examples

### Request
The following is an example of a request.
<!-- {
  "blockType": "request",
  "name": "create_emailthreatsubmission_from_emailthreats"
}
-->
``` http
POST https://graph.microsoft.com/beta/security/threatSubmission/emailThreats
Content-type: application/json

{
  "@odata.type": "#microsoft.graph.emailUrlThreatSubmission",
  "category": "spam",
  "recipientEmailAddress": "tifc@a830edad9050849EQTPWBJZXODQ.onmicrosoft.com",
  "messageUrl": "https://graph.microsoft.com/beta/users/c52ce8db-3e4b-4181-93c4-7d6b6bffaf60/messages/AAMkADU3MWUxOTU0LWNlOTEt="
}
```

> or with tenantAllowOrBlockListAction provided

Reviewer: Lines 70 and 90: Let's create separate examples for separate scenarios. 

The structure should be:
H2 Examples
H3 Example 1: Description
H4 Request
H4 Response

H3 Example 2: Description "with tenantAllowOrBlockListAction provided
H4 Request
... 
Etc.

---
title: "Create emailThreatSubmission"
description: "Create a new emailThreatSubmission object."
author: "caigen"
ms.localizationpriority: medium
ms.prod: "security"
doc_type: apiPageType
---

# Create emailThreatSubmission
Namespace: microsoft.graph.security

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

Create a new [emailThreatSubmission](../resources/security-emailthreatsubmission.md) object.

## Permissions

> One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

Reviewer 2: ```suggestion
One of the following permissions is required to call this API. To learn more permissions, including how to choose one, see [Permissions](/graph/permissions-reference).
```

|Permission type|Permissions (from least to most privileged)|
|:---|:---|
|Delegated (work or school account)|ThreatSubmission.ReadWrite, ThreatSubmission.ReadWrite.All|
|Delegated (personal Microsoft account)|Not supported|
|Application|ThreatSubmission.ReadWrite.All|

## HTTP request

<!-- {
  "blockType": "ignored"
}
-->
``` http
POST /security/threatSubmission/emailThreats
```

## Request headers
|Name|Description|
|:---|:---|
|Authorization|Bearer {token}. Required.|
|Content-Type|application/json. Required.|

## Request body

> In the request body, supply a JSON representation of the [emailContentThreatSubmission](../resources/security-emailcontentthreatsubmission.md) object or the [emailUrlThreatSubmission](../resources/security-emailurlthreatsubmission.md).

Reviewer 2: ```suggestion
In the request body, either supply a JSON representation of the [emailContentThreatSubmission](../resources/security-emailcontentthreatsubmission.md) object or the [emailUrlThreatSubmission](../resources/security-emailurlthreatsubmission.md) object.
```

> The emailContentThreatSubmission and emailUrlThreatSubmission are both emailThreatSubmission but submitted with different ways. 

Reviewer 2: ```suggestion
The emailContentThreatSubmission and emailUrlThreatSubmission are both subtypes of emailThreatSubmission and helps to make submission in different ways. 
```

> * The emailContentThreatSubmission is created with email content itself.

Reviewer 2: ```suggestion
* If you want to create a submission with the email content itself, use the emailContentThreatSubmission.
```

> * The emailUrlThreatSubmissino is created by email url which points to the email. 

Reviewer 2: ```suggestion
* If you want to create a submission with a URL which points to the email, use the emailUrlThreatSubmission.
```

> * After emailContentThreatSubmission and emailUrlThreatSubmision are created, threat submission service itself does not store the email content and just some metadata are stored. So, the creation results are both emailThreatSubmission entity.

Reviewer 2: ```suggestion
* After emailContentThreatSubmission and emailUrlThreatSubmision are created, the threat submission service just stores some metadata about the email. The email content is not stored. So, the created entity from both methods is an emailThreatSubmission entity.
```

---

> title: "Create emailThreatSubmission"

Reviewer 3: Is this the correct file name? There doesn't appear to be a nav property named **emailThreats** on the emailThreatSubmission resource.

---

# Create emailThreatSubmission
Namespace: microsoft.graph.security

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

Create a new [emailThreatSubmission](../resources/security-emailthreatsubmission.md) object.

## Permissions
One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

|Permission type|Permissions (from least to most privileged)|
|:---|:---|
|Delegated (work or school account)|ThreatSubmission.ReadWrite, ThreatSubmission.ReadWrite.All|
|Delegated (personal Microsoft account)|Not supported|
|Application|ThreatSubmission.ReadWrite.All|

## HTTP request

<!-- {
  "blockType": "ignored"
}
-->
``` http
POST /security/threatSubmission/emailThreats
```

## Request headers
|Name|Description|
|:---|:---|
|Authorization|Bearer {token}. Required.|
|Content-Type|application/json. Required.|

## Request body
In the request body, supply a JSON representation of either an [emailContentThreatSubmission](../resources/security-emailcontentthreatsubmission.md) object or an [emailUrlThreatSubmission](../resources/security-emailurlthreatsubmission.md) object.

> The **emailContentThreatSubmission** and **emailUrlThreatSubmission** resources are both subtypes of the **emailThreatSubmission** entity and helps to make submission in different ways. 

Reviewer 3: Can you please clarify the phrase "and helps to make submission in different ways"? Maybe this is what you mean:

"Choose what to include in the request body based on the following:"

> The **emailContentThreatSubmission** and **emailUrlThreatSubmission** resources are both subtypes of the **emailThreatSubmission** entity and helps to make submission in different ways. 

Author: ```suggestion
The **emailContentThreatSubmission** and **emailUrlThreatSubmission** resources are both subtypes of the **emailThreatSubmission** entity. Choose what to include in the request body based on the following:  
```
